extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -120,56 +158,42 @@ void TailFile::onSchedule(const std::shared_ptr<core::ProcessContext> &context,
   std::string value;
 
   if (context->getProperty(Delimiter.getName(), value)) {
-    delimiter_ = value;
+    delimiter_ = parseDelimiter(value);
+  }
+
+  if (!context->getProperty(FileName.getName(), file_to_tail_)) {
+    throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "File to Tail is a required property");
   }
 
   std::string mode;
   context->getProperty(TailMode.getName(), mode);
 
-  std::string file = "";
-  if (!context->getProperty(FileName.getName(), file)) {
-    throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "File to Tail is a required property");
-  }
   if (mode == "Multiple file") {
-    // file is a regex
-    std::string base_dir;
-    if (!context->getProperty(BaseDirectory.getName(), base_dir)) {
+    tail_mode_ = Mode::MULTIPLE;
+
+    if (!context->getProperty(BaseDirectory.getName(), base_dir_)) {
       throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "Base directory is required for multiple tail mode.");
     }
 
-    auto fileRegexSelect = [&](const std::string& path, const std::string& filename) -> bool {
-      if (acceptFile(file, filename)) {
-        tail_states_.insert(std::make_pair(filename, TailState {path, filename, 0, 0}));
-      }
-      return true;
-    };
-
-    utils::file::FileUtils::list_dir(base_dir, fileRegexSelect, logger_, false);
+    // in multiple mode, we check for new/removed files in every onTrigger
 
   } else {
+    tail_mode_ = Mode::SINGLE;
+
     std::string fileLocation, fileName;
-    if (utils::file::PathUtils::getFileNameAndPath(file, fileLocation, fileName)) {
-      tail_states_.insert(std::make_pair(fileName, TailState { fileLocation, fileName, 0, 0 }));
+    if (utils::file::PathUtils::getFileNameAndPath(file_to_tail_, fileLocation, fileName)) {
+      tail_states_.emplace(fileName, TailState{fileLocation, fileName, 0, 0, 0, 0});
```

Comment:
Are we sure about this? 
As far as I understand the change this means that whenever MiNiFi is restarted, the state is dropped and we start reading the file from the beginning.

Reply:
this will be overwritten in `recoverState()` if there is a persisted state, but I agree it looks misleading -- I'll think about how to make it clearer

Reply:
Okay, understood. I'm fine with just having a comment here that tells this.

Reply:
I have added a comment